packages/ui/origin-ui-core/src/components/certificates/CertificatesInbox.tsx

```diff
@@ -0,0 +1,469 @@
+            >
+                <Tab label="Sell" />
+                <Tab label="Withdraw" />
+            </Tabs>
+        );
+    };
+
+    const setEnergyForCertificate = (
+        device: IInboxItemData,
+        cert: IInboxCertificateData,
+        value: number
+    ) => {
+        const newViewData = viewData.map((d) => {
+            if (d.id === device.id) {
+                d.certificates = d.certificates.map((c) => {
+                    if (c.id === cert.id) {
+                        return {
+                            ...c,
+                            energy: value
+                        };
+                    }
+                    return c;
+                });
+            }
+            return d;
+        });
+
+        setViewData(newViewData);
+    };
+
+    const SelectedCertificatesList = (): JSX.Element => {
+        const selectedDeviceCertPairs: [IInboxItemData, IInboxCertificateData][] = [];
+
+        viewData.forEach((d) => {
+            d.certificates.forEach((c) => {
+                if (selectedCerts.includes(c.id)) {
+                    selectedDeviceCertPairs.push([d, c]);
+                }
+            });
+        });
+
+        return (
+            <>
+                {selectedDeviceCertPairs.map(([d, c]) => {
+                    return (
+                        <InboxSelectedItem
+                            key={c.id}
+                            cert={c}
+                            device={d}
+                            setEnergy={(v) => setEnergyForCertificate(d, c, v)}
+                        />
+                    );
+                })}
+
+                {selectedDeviceCertPairs.length === 0 && (
+                    <div className={classes.selectedItemsEmpty}>
+                        {t('certificate.info.selectCertificate')}
+                    </div>
+                )}
+            </>
+        );
+    };
+
+    const user = useSelector(getUserOffchain);
+    const dispatch = useDispatch();
+
+    async function publishForSale() {
+        const certs = getSelectedCertificates();
+
+        certs.forEach((certificate) => {
+            dispatch(
+                requestPublishForSale({
+                    certificateId: parseInt(certificate.id, 10),
+                    amount: BigNumber.from(certificate.energy * 1000000),
+                    price: Math.round((price + Number.EPSILON) * 100),
+                    callback: () => {
+                        updateView();
+                    },
+                    source: certificate.source,
+                    assetId: certificate.assetId
+                })
+            );
+        });
+    }
+
+    async function withdraw() {
+        const certs = getSelectedCertificates();
+        const address = user.blockchainAccountAddress;
+
+        certs.forEach((certificate) => {
+            const assetId = certificate.assetId;
+            const amount = (certificate.energy * 1000000).toString();
```

Comment:
If I am not mistaken such numbers (`* 1000000`) should be received from some external source, because the energy is configurable and hard-coded energy transformation values can cause bugs in future.
There should be examples of getting this number throughout code or even an existing helper function like `energyFormatter`